### PR TITLE
Att mask

### PR DESCRIPTION
**Pull request by glkuzi**
Small bugfixes:

- added attention mask to the BERT input
- changed the number of frozen layers in BERT based models from 10 to 12
- added mapping to ints for the gender label in the BIOS dataset
- added missed dependencies
- changed best model selection from val loss criterion to val accuracy

### CODE REVIEW

**Comment by HanXudong** [comment]
Hi Gleb,

Many thanks for your comments again! 

Following our previous discussions, I've made the following changes:

- update the Python version from 3.7 -> 3.8, and the NumPy version from 1.21 to 1.22 for security reasons. See https://github.com/HanXudong/fairlib/security/dependabot/2 for details.
- include the attention mask as input to BERT-based models
- replace tuples with dictionaries for each minibatch iteration
- update the classifiers and discriminators (for adversarial learning) to support the additional inputs 
- include the label mapping for BIOS dataset

I will then close the pull request. 

Best,
Xudong
